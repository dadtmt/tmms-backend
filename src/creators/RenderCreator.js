import R from 'ramda'
import React from 'react'
import { branch, renderComponent } from 'recompose'

import SheetSelector from './SheetSelectorWithData'
import Dice from './Dice'

const BaseComponent = () => <div />

const mayRenderDice = branch(
  R.propEq('type', 'dice'),
  renderComponent(Dice)
)

const mayRenderCharacterSheet = branch(
  R.propEq('type', 'characterSheet'),
  renderComponent(SheetSelector)
)

export default R.pipe(mayRenderCharacterSheet, mayRenderDice)(BaseComponent)
