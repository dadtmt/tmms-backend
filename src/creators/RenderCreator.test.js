import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { reducer } from 'redux-form'

import RenderCreator from './RenderCreator'

const store = createStore(reducer)

it('renders without crashing', () => {
  const div = document.createElement('div')

  ReactDOM.render(
    <Provider store={store}><RenderCreator /></Provider>,
    div
  )
})

it('renders dice if dice type', () => {
  expect(
    renderer.create(<Provider store={store}>
      <RenderCreator type='dice' />
    </Provider>).toJSON()
  ).toMatchSnapshot()
})
