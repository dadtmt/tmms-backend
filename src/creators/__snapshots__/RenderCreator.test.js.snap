exports[`test renders dice if dice type 1`] = `
<form>
  <div
    className="checkbox"
    style={undefined}>
    <label>
      <input
        checked=""
        disabled={false}
        name="master"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="checkbox"
        value="" />
      The story teller rolls the dice!
    </label>
  </div>
  <div
    className="form-group">
    <label
      className="control-label"
      htmlFor="nbDices">
      Number of dices:
    </label>
    <input
      className="form-control"
      id="nbDices"
      name="nbDices"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder={undefined}
      type="number"
      value="" />
    <span
      className="help-block" />
  </div>
  <div
    className="form-group">
    <label
      className="control-label"
      htmlFor="nbSides">
      Number of sides:
    </label>
    <input
      className="form-control"
      id="nbSides"
      name="nbSides"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder={undefined}
      type="number"
      value="" />
    <span
      className="help-block" />
  </div>
  <div
    className="form-group">
    <label
      className="control-label"
      htmlFor="modifier">
      Modifier:
    </label>
    <input
      className="form-control"
      id="modifier"
      name="modifier"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder={undefined}
      type="number"
      value="" />
    <span
      className="help-block" />
  </div>
</form>
`;
